player/command.c

```diff
@@ -7704,216 +7703,247 @@ void mp_option_change_callback(void *ctx, struct m_config_option *co, int flags,
     if (self_update)
         return;
 
-    if (flags & UPDATE_TERM)
-        mp_update_logging(mpctx, false);
-
-    if (flags & (UPDATE_OSD | UPDATE_SUB_FILT | UPDATE_SUB_HARD)) {
-        for (int n = 0; n < num_ptracks[STREAM_SUB]; n++) {
-            struct track *track = mpctx->current_track[n][STREAM_SUB];
-            struct dec_sub *sub = track ? track->d_sub : NULL;
-            if (sub) {
-                int ret = sub_control(sub, SD_CTRL_UPDATE_OPTS,
-                                      (void *)(uintptr_t)flags);
-                if (ret == CONTROL_OK && flags & (UPDATE_SUB_FILT | UPDATE_SUB_HARD)) {
-                    sub_redecode_cached_packets(sub);
-                    sub_reset(sub);
-                    if (track->selected)
-                        reselect_demux_stream(mpctx, track, true);
+    // Coalesce redundant updates and only keep the newest one.
+    bool drop = false;
+    for (int i = 0; i < mpctx->num_option_callbacks; i++) {
+        if (flags && flags == mpctx->option_callbacks[i].flags)
+            drop = true;
+        if (mpctx->option_callbacks[i].co && opt_ptr == mpctx->option_callbacks[i].co->data)
+            drop = true;
+        if (drop && i < mpctx->num_option_callbacks - 1)
```

Comment:
You can do `if (!drop) continue;` here, to avoid checking it twice again.

Generally I think it make sense, probably need more testing and likely not so big impact in normal use of mpv.